projects/igniteui-angular/src/lib/icon/filtering-icons.ts

```diff
@@ -0,0 +1,44 @@
+import { InjectionToken, inject } from '@angular/core';
+import { IMXIcon, contains, doesNotContain, endsWith, equals, greaterThan, greaterThanOrEqual, isAfter, isBefore, isEmpty, isFalse, isNull, isNotNull, isTrue, lastMonth, lastYear, lessThan, lessThanOrEqual, nextMonth, nextYear, notEmpty, notEqual, selectAll, startsWith, thisMonth, thisYear, today, ungroup, yesterday } from '@igniteui/material-icons-extended';
+import { IgxIconService } from './icon.service';
+
+export const filteringIcons: IMXIcon[] = [
+    contains,
+    doesNotContain,
+    endsWith,
+    equals,
+    greaterThan,
+    greaterThanOrEqual,
+    isAfter,
+    isBefore,
+    isEmpty,
+    isFalse,
+    isNotNull,
+    isNull,
+    isTrue,
+    lastMonth,
+    lastYear,
+    lessThan,
+    lessThanOrEqual,
+    nextMonth,
+    nextYear,
+    notEmpty,
+    notEqual,
+    selectAll,
+    startsWith,
+    thisMonth,
+    thisYear,
+    today,
+    ungroup,
+    yesterday
+];
+export const FILTERING_ICONS = new InjectionToken<void>('FILTERING_ICONS', {
```

Comment:
Okay, not exactly what I had in mind, but could consider.
First issue with the token approach is it doesn't tree-shake without explicit `/*@__PURE__*/` before it (see others in the repo) and thus drags the factory function and the icons collection with the exports from the package, thus making those permanently in the bundle. See the bundle app build warning here:
https://github.com/IgniteUI/igniteui-angular/actions/runs/11162862017/job/31028455568#step:16:63
(just FYI same thing happens w/ just a chip component active).

The other issue with this is that due to this being injected (thus on component init) - these icons will be registered even when not in use, so again might be better as a simpler reusable function.

Also no reason for `filteringIcons` and `registerFilteringSVGIconsFactory` to be exported as far as I can tell.

Reply:
There's also something going on with the extended icons - I get all of them bundled, which makes no sense and there might be an additional issue there as well.

Reply:
Okay, seems to be to do with the `all` export. If split like this:
```ts
import {
    contains, doesNotContain, endsWith, equals, greaterThan, greaterThanOrEqual, isAfter, isBefore,
    isEmpty, isFalse, isNull, isNotNull, isTrue, lastMonth, lastYear, lessThan, lessThanOrEqual, nextMonth,
    nextYear, notEmpty, notEqual, selectAll, startsWith, thisMonth, thisYear, today, ungroup, yesterday
} from '@igniteui/material-icons-extended/editor';

import { type IMXIcon } from '@igniteui/material-icons-extended';
```
Tree-shaking seems to work fine